### ISSUE TITLE
[REC] Subtlety Rogue recommends Eviscerate rather than Slice and Dice, and instead recommends Slice and Dice in cooldown window 

### ISSUE THREAD

**Issue by fzarr** [Hekili/hekili#1525]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

Outside of dance and symbols, slice and dice was about to fall off and I expected a recommendation to refresh it. However it recommends eviscerate instead, and then in the future list of actions I can see a recommendation to use dance, shadowstrike x2, then slice and dice which to me seems less efficient and not what the APL would suggest (eviscerate being the last finisher in the list).

I see in the snapshot, there is a 'refreshable' flag associated with slice and dice that gets set to false when it feels it should be true, but not sure I am reading it correctly.

### How to Reproduce

I couldn't reproduce the exact scenario I had in the snapshot easily, but a similar situation is as follows:

1. Apply Slice and Dice (can be low duration for testing)
2. Use cooldowns and /cancelaura them
3. Build to 5 cps
4. Rupture
5. Build to 5 cps
6. Slice and Dice will only be recommended to be re-applied when it completely falls off or is 1-2s from falling off (even if initial snd was applied with 5-6 cps and thus should have a 10-12s pandemic window)

### Snapshot (Link)

https://pastebin.com/zCNaJgZq

**Comment by Hekili** [comment]
Thanks, this was a good report and a well-chosen snapshot.

It looks like I have erroneously based `refreshable` on the maximum duration of the buffs/debuffs instead of based on the duration of the DOT that would be applied now, with your current Combo Points.  I also needed to update Slice and Dice to adjust its expected duration based on your current `effective_combo_points`.

In your snapshot, you had 3.87s of Slice and Dice left.  You had 5 combo points, and Slice and Dice would apply for 36 seconds with 5 combo points.  `refreshable` should be `true` when the remaining duration of a buff/debuff is less than or equal to 30% of the newly applied duration.  That'd be 10.8 seconds.  Refreshable should've been true, but Slice and Dice's application duration was incorrectly set to 10 seconds.  Which meant the 3.87 remaining seconds was greater than 0.3 * 10 = 3 seconds, which meant it wasn't considered refreshable.

```
 - this entry's criteria FAILS: ! variable.premed_snd_condition[false] & spell_targets.shuriken_storm[1.00] < 6 & ! buff.shadow_dance.up[false] & buff.slice_and_dice.remains[3.87] < fight_remains[1647.00] & refreshable[false]
```

The addon did attempt to figure out if Slice and Dice would be used when the buff was "refreshable" (< 3 seconds), and was ready to recommend that at 0.88 seconds, but Eviscerate was available sooner.

```
Recheck #2 ( +0.88 ) MET: ! variable.premed_snd_condition[false] & spell_targets.shuriken_storm[1.00] < 6 & ! buff.shadow_dance.up[false] & buff.slice_and_dice.remains[2.99] < fight_remains[1646.12] & refreshable[true]
```

I'm posting an alpha build now, where I've rechecked all of the buffs/debuffs that are affected by combo points to make their expected apply duration match in-game behavior.  The result should be that SnD would've been refreshed in the first slot of your snapshot.

https://github.com/Hekili/hekili/releases/tag/v9.2.5-1.0.1-alpha4
